### PR TITLE
Use one generic pod template for tests

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Exposing a simple interface to users adding tests requires that we DRY
out the boilerplate that we use to run them. The first step is to use
one template to create the Pods we run the tests in. These can simply be
parameterized with the name of the test, command-line to run and the
quota allocation.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @csrwng this is the first bubble

### CODE REVIEW

File: cluster/ci/origin/config/tests.yml [hunk 1]
```diff
@@ -3,6 +3,26 @@ kind: Template
 labels:
   template: integration-tests
```
Comment (csrwng) [inline cluster/ci/origin/config/tests.yml]: s/integration-tests/test/

Comment (stevekuznetsov) [reply cluster/ci/origin/config/tests.yml]: whoops, done

**Comment by stevekuznetsov** [comment]
Unclear if the `$REPO_OWNER` etc defaults make sense if this is truly generic but we can tackle it when we cross that boundary

**Comment by stevekuznetsov** [comment]
Test is here: https://jenkins-origin-ci.svc.ci.openshift.org/job/steve-ci-steve-origin/4

Pod is here: pods/cmd-4-pr-16839-e3a7a904cd24b3a640170398-f34m0

Failing to schedule atm

**Comment by stevekuznetsov** [comment]
Podspec looks good, will merge:

```
$ oc get pod cmd-4-pr-16839-e3a7a904cd24b3a640170398-f34m0 -o jsonpath='{.spec.containers[0].command}'
[release-ci entrypoint --config-path=/var/run/logging-config/logging-config.json -- /bin/bash -c env HOME=/tmp HOSTNAME=$(hostname) TERM=xterm-256color OS_OUTPUT_SCRIPTPATH=/var/run/logging JUNIT_REPORT=true SKIP_TEARDOWN=1 make test-cmd]
```

**Pull request merged**